tap_github/authenticator.py

```diff
@@ -196,18 +213,21 @@ def __init__(self, stream: RESTStream) -> None:
         self.logger: logging.Logger = stream.logger
         self.tap_name: str = stream.tap_name
         self._config: Dict[str, Any] = dict(stream.config)
-        self.tokens_map = self.prepare_tokens()
-        self.active_token: Optional[TokenRateLimit] = (
-            choice(list(self.tokens_map.values())) if len(self.tokens_map) else None
+        self.token_managers = self.prepare_tokens()
+        self.active_token: Optional[TokenManager] = (
+            choice(self.token_managers) if len(self.token_managers) else None
         )
 
     def get_next_auth_token(self) -> None:
-        tokens_list = list(self.tokens_map.items())
+        token_managers = self.token_managers
         current_token = self.active_token.token if self.active_token else ""
-        shuffle(tokens_list)
-        for _, token_rate_limit in tokens_list:
-            if token_rate_limit.is_valid() and current_token != token_rate_limit.token:
-                self.active_token = token_rate_limit
+        shuffle(token_managers)
```

Comment:
This reorders the list in place, right? Does that affect the `self.token_managers`? Is that the intended behavior?

Totally OK if it's _yes_ to all, just wanna double check 🙏

Reply:
Good catch!  The order of `self.token_managers` shouldn't matter in theory since the order isn't currently used anywhere (until now it was a dict after all), but changing it also isn't intentional and could trip up a future dev.  I'll confirm whether `self.token_managers` is getting changed and tweak if so. 🙏

Reply:
Yup, it was shuffling `self.token_manager`.  Pushed a commit adding a `deepcopy`.

Reply:
Awesome, thanks for confirming and updating!